internal/pkg/service/common/etcdop/op/txn.go

```diff
@@ -368,6 +368,33 @@ func (v *lowLevelTxn[R]) addElse(op etcd.Op, mapper MapFn) {
 }
 
 func (v *lowLevelTxn[R]) mergeTxn(ctx context.Context, op Op) error {
+	// Step down to a nested Merge operation, if there is no processor/else branch.
+	if txn, ok := op.(txnInterface); ok && isSimpleTxn(txn) {
+		thenStart := len(v.thenOps)
+		elseStart := len(v.elseOps)
+		if err := v.addParts(ctx, txn); err != nil {
+			return err
+		}
+
+		if txn.txnProcessorsCount() > 0 {
+			thenEnd := len(v.thenOps)
+			elseEnd := len(v.elseOps)
+			v.processors = append(v.processors, func(ctx context.Context, r *TxnResult[R]) {
+				response := &etcd.TxnResponse{Succeeded: r.Succeeded()}
+				if r.Succeeded() {
+					response.Responses = r.Response().Txn().Responses[thenStart:thenEnd]
+				} else {
+					response.Responses = r.Response().Txn().Responses[elseStart:elseEnd]
+				}
+
+				// Call original mapper of the sub transaction
+				txn.txnInvokeProcessors(ctx, r.resultBase)
+			})
+		}
+
+		return nil
+	}
+
```

Comment:
:arrow_right: :arrow_right: :arrow_right:  Implementation of the fix. Transaction without `if/else` parts can be directly processed.